File: 225. Implement Stack using Queues.py
class Stack(object):
    def __init__(self):
        """
        initialize your data structure here.
        """
        self.q1 = []
        self.q2 = []

    def push(self, x):
        """
        :type x: int
        :rtype: nothing
        """
        self.q1 += [x]

    def pop(self):
        """
        :rtype: nothing
        """
        while len(self.q1)>1:
            self.q2 += [self.q1.pop(0)]
        self.q1.pop(0)
        while self.q2:
            self.q1 += [self.q2.pop(0)]

    def top(self):
        """
        :rtype: int
        """
        while len(self.q1)>1:
            self.q2 += [self.q1.pop(0)]
        temp = self.q1.pop(0)
        self.q2 += [temp]
        while self.q2:
            self.q1 += [self.q2.pop(0)]
        return temp

    def empty(self):
        """
        :rtype: bool
        """
        return False if self.q1 else True

File: 225. Implement Stack using Queues.js
/**
 * @constructor
 */
var Stack = function() {
    this.q1 = [];
    this.q2 = [];
};

/**
 * @param {number} x
 * @returns {void}
 */
Stack.prototype.push = function(x) {
    this.q1.push(x);
};

/**
 * @returns {void}
 */
Stack.prototype.pop = function() {
    while (this.q1.length > 1)
        this.q2.push(this.q1.shift());
    this.q1.shift();
    while (this.q2.length > 0)
        this.q1.push(this.q2.shift());
};

/**
 * @returns {number}
 */
Stack.prototype.top = function() {
    while (this.q1.length > 1)
        this.q2.push(this.q1.shift());
    var temp = this.q1.shift();
    this.q2.push(temp);
    while (this.q2.length > 0)
        this.q1.push(this.q2.shift());
    return temp;
};

/**
 * @returns {boolean}
 */
Stack.prototype.empty = function() {
    return this.q1.length === 0 ? true : false;
};
